ContentSecurityPolicy/Violation/Report.php

```diff
@@ -146,6 +153,6 @@ public static function fromRequest(Request $request)
             $ret['script-sample'] = $report['csp-report']['script-sample'];
         }
 
-        return new self($report);
+        return new self($report, $request->headers->get('User-Agent'));
```

Comment:
I saw this the other day (not related to this change), just asking here because I'm not sure about the array shape.

https://github.com/nelmio/NelmioSecurityBundle/blob/8f2c69a6439f4fdaa320462ceadeb1f052b69cbe/src/ContentSecurityPolicy/Violation/Report.php#L127-L151

here the first `if` looks weird and looks like the json file should be like 

```
[
  'csp-report' => [
    'csp-report' => [
      'effective-directive' => '...'
    ]
  ]
]
```
and then `$ret` variable is not used.

I wanted to take a look at this, I guessed that the "right" thing would be to change the `if` like:

```php
if (!is_array($report) || [] === $report)
```

and replace `$report['csp-report']` by `$report` and then if it's not necessary, we can remove the `$ret` variable or make it like `$report` with the current replacements.